### ISSUE TITLE
Process `bam_ingress:minimap2_alignment (1)` terminated with an error exit status (137)

### ISSUE THREAD

**Issue by Yoon90** [epi2me-labs/wf-human-variation#133]
### Ask away!

Hello!
[nextflow.log](https://github.com/epi2me-labs/wf-human-variation/files/14007577/nextflow.log)

Human-variation-workflow has stopped with an error exit status(137)
I guess it has occurred because of the shortage of RAM.

I'm attaching the head and the middle part of the nextflow.config

I found that the minimal required memory is 32GB but this computer cannot provide that much.
Is there any option to change/lower required Memory(RAM)?

Logs:

[0;92m||||||||||   [0m[2m_____ ____ ___ ____  __  __ _____      _       _
[0;92m||||||||||  [0m[2m| ____|  _ \_ _|___ \|  \/  | ____|    | | __ _| |__  ___
[0;33m|||||       [0m[2m|  _| | |_) | |  __) | |\/| |  _| _____| |/ _` | '_ \/ __|
[0;33m|||||       [0m[2m| |___|  __/| | / __/| |  | | |__|_____| | (_| | |_) \__ \
[0;94m||||||||||  [0m[2m|_____|_|  |___|_____|_|  |_|_____|    |_|\__,_|_.__/|___/
[0;94m||||||||||  [0m[1mwf-human-variation v1.9.2[0m
[2m--------------------------------------------------------------------------------[0m
[1mCore Nextflow options[0m
  [0;34mrunName         : [0;32mtender_pike[0m
  [0;34mcontainerEngine : [0;32mdocker[0m
  [0;34mcontainer       : [0;32montresearch/wf-human-variation:sha0800eade05e4cbb75d45421633c78c4f6320b2f6[0m
  [0;34mlaunchDir       : [0;32m/home/heeji/epi2melabs/instances/wf-human-variation_01HMQ5WY7W4NNY8QABXDCK57S2[0m
  [0;34mworkDir         : [0;32m/home/heeji/epi2melabs/instances/wf-human-variation_01HMQ5WY7W4NNY8QABXDCK57S2/work[0m
  [0;34mprojectDir      : [0;32m/home/heeji/epi2melabs/workflows/epi2me-labs/wf-human-variation[0m
  [0;34muserName        : [0;32mheeji[0m
  [0;34mprofile         : [0;32mstandard[0m
  [0;34mconfigFiles     : [0;32m/home/heeji/epi2melabs/workflows/epi2me-labs/wf-human-variation/nextflow.config[0m

[1mWorkflow Options[0m
  [0;34msv              : [0;32mtrue[0m
  [0;34msnp             : [0;32mtrue[0m
  [0;34mcnv             : [0;32mtrue[0m
  [0;34mstr             : [0;32mtrue[0m

[1mMain options[0m
  [0;34msample_name     : [0;32mDBS_barcode05[0m
  [0;34mbam             : [0;32m/media/heeji/2023_WES/nanopore/231227_DBS_4_pilot_test/no_sample/20231227_1908_X5_FAU99700_eaea91de/bam_pass/barcode05/merge_barcode05.bam[0m
  [0;34mref             : [0;32m/media/heeji/2023_WES/hg38.fa[0m
  [0;34mbam_min_coverage: [0;32m5[0m
  [0;34mout_dir         : [0;32m/home/heeji/epi2melabs/instances/wf-human-variation_01HMQ5WY7W4NNY8QABXDCK57S2/output[0m

!! Only displaying parameters that differ from the pipeline defaults !!
[2m--------------------------------------------------------------------------------[0m
If you use epi2me-labs/wf-human-variation for your analysis please cite:

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

[2m--------------------------------------------------------------------------------[0m
This is epi2me-labs/wf-human-variation v1.9.2.

Caused by:
  Process `bam_ingress:minimap2_alignment (1)` terminated with an error exit status (137)

Command executed:

  samtools bam2fq -@ 1 -T 1  merge_barcode05.bam | minimap2 -y -t 8 -ax map-ont hg38.fa -     | samtools sort -@ 3 --write-index -o DBS_barcode05.bam##idx##DBS_barcode05.bam.bai -O bam --reference hg38.fa -

Command exit status:
  137

Command output:
  (empty)

Command error:
  [M::mm_idx_gen::116.452*1.80] collected minimizers
  [M::mm_idx_gen::217.785*1.78] sorted minimizers
  [M::main::217.792*1.78] loaded/built the index for 455 target sequence(s)
  [M::mm_mapopt_update::219.462*1.78] mid_occ = 728
  [M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 455
  [M::mm_idx_stat::220.401*1.77] distinct minimizers: 100202295 (37.96% are singletons); average occurrences: 5.732; average spacing: 5.587; total length: 3209286105
  [M::bam2fq_mainloop] discarded 0 singletons
  [M::bam2fq_mainloop] processed 236000 reads
  [bam_sort_core] merging from 0 files and 3 in-memory blocks...
  .command.sh: line 2:    25 Done                    samtools bam2fq -@ 1 -T 1 merge_barcode05.bam
          26 Killed                  | minimap2 -y -t 8 -ax map-ont hg38.fa -
          28                       | samtools sort -@ 3 --write-index -o DBS_barcode05.bam##idx##DBS_barcode05.bam.bai -O bam --reference hg38.fa -

**Comment by RenzoTale88** [comment]
@Yoon90 in addition to what @SamStudio8 mentioned, you can provide customized memory solutions with a custom profile. For instance, to increase/decrease the minimap2 memory, you can create a file called `custom_memory.config`, with the following code inside it:
```
process {
    withName "minimap2_alignment" {
        memory = "64.GB"
    }
}
```
Or any memory specification you think is appropriate. You can then provide this with `-c custom_memory.config`.

**Comment by cjw85** [comment]
You are correct, the workflow needs a minimum of 32Gb RAM to run, and very often the peak memory usage will be more than this. If your computer has only 32Gb RAM this workflow is unlikely to run reliably.
